default/python/AI/Behavior/Character.py

```diff
@@ -0,0 +1,517 @@
+        """Select a fraction less than 1 at maximum ratio of PP for buildings"""
+        return None
+
+    def preferred_discount_multiplier(self, alternatives):  # pylint: disable=no-self-use,unused-argument
+        """Select a discount multiplier for use in evaluate planet"""
+        return None
+
+    def max_defense_portion(self):  # pylint: disable=no-self-use,unused-argument
+        """Return maximum fraction of production for defense"""
+        return 1.0
+
+    def check_orbital_production(self):  # pylint: disable=no-self-use,unused-argument
+        """Return if orbital production is checked this turn"""
+        return False
+
+    def target_number_of_orbitals(self):  # pylint: disable=no-self-use,unused-argument
+        """Return target number of orbitals"""
+        return 0
+
+    def may_build_building(self, building):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to build ''building''"""
+        return True
+
+    def may_produce_troops(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to produce troop ships"""
+        return True
+
+    def military_safety_factor(self):  # pylint: disable=no-self-use,unused-argument
+        """Return military safety factor"""
+        return 0.0
+
+    def may_dither_focus_to_gain_research(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to trade production at a loss for research"""
+        return True
+
+    def may_research_heavily(self):  # pylint: disable=no-self-use,unused-argument
+        """Return true if allowed to target research/industry > 1.5"""
+        return True
+
+    def may_travel_beyond_supply(self, distance):  # pylint: disable=no-self-use,unused-argument
+        """Return True if able to travel distance hops beyond empire supply"""
+        return True
+
+    def get_research_index(self):  # pylint: disable=no-self-use,unused-argument
+        """Deprecated"""
+        return None
+
+    def may_research_xeno_genetics_variances(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if able to vary xeno genetics research"""
+        return True
+
+    def prefer_research_defensive(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if should prefer defensive tech research"""
+        return True
+
+    def prefer_research_low_aggression(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if should prefer less aggressive tech research"""
+        return True
+
+    def may_research_tech(self, tech):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to research ''tech''"""
+        return True
+
+    def may_research_tech_classic(self, tech):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to research ''tech'' in the classic algorithm"""
+        return True
+
```

Comment:
If you can not give more descriptive docstrings because you do not know the underlying code, that is no problem and expected. Simply refering to the place(s) where the function is invoked would be already a great help, e.g. "Toggles Bla in Research.py" is more helpful than  "Return True if able to vary xeno genetics research" with no indication to whatever that is and whereever it is used.
More descriptive docstrings can be added later, TODOs would not hurt though.